openapi.yaml

```
@@ -555,6 +555,463 @@ paths:
+                                 api_key="<API-TOKEN>",
+                                 ib_context="<IB-CONTEXT>")
+
+                  # Get file size
+                  local_filepath = 'large-file.pdf'
+                  filename = 'large-file.pdf'
+                  file_size = os.path.getsize(local_filepath)
+
+                  # Create multipart upload session
+                  session_response = client.batches.create_multipart_upload_session(
+                      batch_id=123456,
+                      filename=filename,
+                      file_size=file_size
+                  )
+
+                  session_id = session_response.session_id
+                  part_size = session_response.part_size
+                  print(f"Session ID: {session_id}")
+                  print(f"Part size: {part_size}")
+            - sdk: python
+              name: without SDK
+              code: |
+                import requests
+                import json
+                import os
+                
+                # Get file size
+                local_filepath = 'large-file.pdf'
+                filename = 'large-file.pdf'
+                size = os.path.getsize(local_filepath)
+                
+                url = "https://aihub.instabase.com/api/v2/batches/multipart-upload"
+                headers = { "Authorization": "Bearer abcdefghijklmnopqrst1234567890",
+                            "IB-Context": 'john.doe_acme.com',
+                            "Content-Type": "application/json" }
+                
+                # create the request payload
+                data = { "batch_id": 123456,
+                         "filename": filename,
+                         "file_size": str(size) }
+                
+                # make the POST request
+                response = requests.post(url,
+                                         headers=headers,
+                                         json=data)
+                
+                # handle the response
+                if response.status_code == 201:
+                    part_size = response.json()['part_size']
+                    session_id = response.json()['session_id']
+                    print(f"Session ID: {session_id}")
+                    print(f"Part size: {part_size}")
+                else:
+                    print(f"Error: {response.status_code} - {response.text}")
+
+  /v2/batches/multipart-upload/sessions/{session_id}/parts/{part_num}:
+    put:
+      operationId: uploadMultipartPart
+      x-fern-audiences:
+        - public
+      tags:
+        - batches
+      summary: Upload part to multipart session
+      description: |
+        Upload part of a file to the multipart upload session. Each part's size should match the `part_size` returned by the [Start multipart upload session](/api-sdk/api-reference/batches/create-multipart-upload-session) call, except for the final part, which can be smaller.
```

```suggestion
        Upload part of a file to the multipart upload session. The size of each part must match the `part_size` returned by the [Start multipart upload session](/api-sdk/api-reference/batches/create-multipart-upload-session) call, except for the final part, which can be smaller.
```

For some reason the other people on the docs team have a prohibition on possessives. I don’t get why, but whatevs.